compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2341,6 +2342,16 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
 
         setTypeOfVarRef(tupleDeStmt.varRef, data);
 
+        if (tupleDeStmt.expr.getKind() == LIST_CONSTRUCTOR_EXPR) {
```

Comment:
We have avoided having record literals for record destructures by logging an error. We had a live discussion on this and came to conclusion to avoid having record literals within a list destructure for scenarios with mapping BP within list BP.

Reply:
But, as a second thought, we should allow this.